include/Gaffer/ComputeNode.h

```diff
@@ -83,6 +89,21 @@ class GAFFER_API ComputeNode : public DependencyNode
 		virtual ValuePlug::CachePolicy hashCachePolicy( const ValuePlug *output ) const;
 		/// Called to determine how calls to `compute()` should be cached. If `compute( output )`
 		/// will spawn TBB tasks then one of the task-based policies _must_ be used.
+		///
+		/// > Caution : Using any policy other than `Uncached` or `Legacy` places constraints
+		/// > on the implementation of `hash()` : the hash _must be different_ to the hash of
+		/// > any upstream dependency of `compute()`. In practice, this means two things :
+		/// >
+		/// > 1. A direct pass-through of an input hash must not be used. See
+		/// >    `ObjectProcessor::processedObjectPlug()` for an example
+		/// >    of using TaskCollaboration while retaining the ability to use
+		/// >    pass-throughs.
```

Comment:
Am I just missing something, or is this something that hasn't been implemented yet?  It's not obvious from looking at ObjectProcessor::processedObjectPlug how it is used with TaskCollaboration ... perhaps some subclasses turn it on?

Reply:
This is implemented already, with OSLObject being one example because it returns TaskCollaboration from `processedObjectComputeCachePolicy(). But `ObjectProcessor::computeObject()` can still do a perfect pass-through when disabled, because the `out.object` uses the default policy, and only pulls on `__processedObject` when the node is enabled.